pandas/plotting/_matplotlib/core.py

```diff
@@ -1231,16 +1231,22 @@ def _make_plot(self):
         else:
             c_values = c
 
-        # cmap is only used if c_values are integers, otherwise UserWarning
-        if is_integer_dtype(c_values):
-            # pandas uses colormap, matplotlib uses cmap.
-            cmap = self.colormap or "Greys"
+        if self.colormap is not None:
             if mpl_ge_3_6_0():
-                cmap = mpl.colormaps[cmap]
+                cmap = mpl.colormaps[self.colormap]
             else:
-                cmap = self.plt.cm.get_cmap(cmap)
+                cmap = self.plt.cm.get_cmap(self.colormap)
         else:
-            cmap = None
+            # cmap is only used if c_values are integers, otherwise UserWarning
```

Comment:
@phofl fyi I think this caused a new warning in `test_scatter_with_c_column_name_with_colors`

```
pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_scatter_with_c_column_name_with_colors[c-Greys]
pandas/tests/plotting/frame/test_frame_color.py::TestDataFrameColor::test_scatter_with_c_column_name_with_colors[color-Greys]
  /Users/runner/work/pandas/pandas/pandas/plotting/_matplotlib/core.py:1270: UserWarning: No data for colormapping provided via 'c'. Parameters 'cmap' will be ignored
    scatter = ax.scatter(
```

Does that mean we should expect the UserWarning?

Reply:
Yes, had a different version installed locally.